tests/unit/amo/components/TestAddonBadges.js

```diff
@@ -181,4 +181,24 @@ describe(__filename, () => {
       expect(root.find(Badge)).toHaveLength(0);
     });
   });
+
+  it('displays a badge when the addon requires payment', () => {
+    const addon = createInternalAddon(createFakeAddon({
+      requires_payment: true,
+    }));
+    const root = shallowRender({ addon });
+
+    expect(root.find(Badge)).toHaveProp('type', 'requires-payment');
+    expect(root.find(Badge))
+      .toHaveProp('label', 'Some features may require payment');
+  });
+
+  it('does not display a badge when the addon does not require payment', () => {
+    const addon = createInternalAddon(createFakeAddon({
+      requires_payment: false,
+    }));
+    const root = shallowRender({ addon });
+
+    expect(root.find(Badge)).toHaveLength(0);
```

Comment:
This is fine (and probably mirrors other tests), but technically we just want no badges with the "requires payment" prop.

This might be a bit more accurate as `expect(root.find(Badge).find({ type: 'requires-payment' })).toHaveLength(0);`, but obviously it works as-is. Just means if we added a `Badge` elsewhere we'd need to change this test, making it less atomic 😄

Reply:
This is a good point. I changed this test, and the others that are asserting the absence of a particular badge. I also added a test that checks that no badges at all are present when none are called for. This would break if we did add a different `Badge` element somewhere inside the `Addon`, but if we do that we'd probably need to revisit this test anyway.

Perhaps you want to take a final look at this before I merge it, @tofumatt?